source/parser.hera

```diff
@@ -3539,6 +3539,15 @@ PropertyDefinition
   # NOTE: basic identifiers are now part of the rule above
   #_?:ws IdentifierReference:id ->
   #  return prepend(ws, id)
+  InsertDotDotDot:dots IterationExpression:exp ->
+    exp.statement.object = true
```

Comment:
We shouldn't modify objects like this in `parser.hera` (only in postprocessing steps), because it can mess up caching. Please change to be an immutable update, like
```js
let { statement } = exp
statement = { ...statement, object: true }
exp = {
  ...exp,
  statement,
  children: exp.children.map(($) => $ === exp.statement ? statement : $),
}
```

Reply:
This will also generate weird code when the iteration is a reduction, given that it takes priority over `object` behavior. Perhaps we should raise an error in this case? Something like
```js
const children = [ dots, exp ]
if (exp.statement.reduction) {
  children.unshift({
    type: "Error",
    message: "Reduction loops are forbidden in object literals",
  })
}
```

Alternatively, the error detection could be done in `iterationDeclaration` below.